public/index.js

```diff
@@ -0,0 +1,51 @@
+(function () {
+  var signUpForm = document.querySelector('.signup-form');
+  var signUpButton = document.querySelector('.signup-button');
+  signUpForm.addEventListener('submit', function (e) {
+    e.preventDefault();
+    signUpButton.disabled = true;
+    var elements = signUpForm.elements;
+    var reqBody = {};
+    for (var i = 0; i < elements.length; i++) {
+      var input = elements[i];
+      var name = input.dataset.name || input.name;
+      var type = input.dataset.type || input.type;
+      if (!name || !type) continue;
+      switch (type) {
+        case 'select-one':
+        case 'text':
+          reqBody[name] = input.value;
+          break;
+        case 'number':
+          reqBody[name] = Number(input.value);
+          break;
+        default:
+          alert('Cannot handle input type: ' + type + ' for input: ' + name);
+          break;
+      }
+    }
+    var xhr = new XMLHttpRequest();
+    xhr.open("POST", "https://covid-spotter.web.app/subscribe");
```

Comment:
Or we should just make it relative since the API won't work cross-origin any way.

```suggestion
    xhr.open("POST", "/subscribe");
```

Reply:
Yeah, that's a good idea